beps/0003-auth-architecture-evolution/README.md

```diff
@@ -97,25 +99,105 @@ export interface AuthService {
 }
 ```
 
-#### Usage Patterns
+### `AuthService` Usage Patterns
 
 TODO
 
-### `HttpAuthService` (WIP)
+### `HttpRouterService` Interface
 
-The new `HttpAuthService` interface is defined as follows:
+> Open question: Should this instead be added to the `HttpAuthService`? It may fit a bit better there, but on the other hand it might make sense to add additional policies unrelated to authentication too, such as rate limiting.
 
-```ts
-type AuthTypes = 'user' | 'user-cookie' | 'service' | 'unauthorized';
+The `HttpRouterService` interface will be extended with the ability to opt-out of the default protection of endpoints, enabling either cookie auth or unauthenticated access.
 
-export interface HttpAuthServiceMiddlewareOptions {
-  allow: AuthTypes[];
+```ts
+export interface HttpRouterService {
+  // All routes only allow authenticated users and services by default.
+  use(handler: Handler): void;
+
+  // Exact option structure is TBD, just highlighting the general idea for now
+  configure(options: {
+    allowCookieAuthOnPaths?: string[];
+    allowUnauthenticatedAccessPaths?: string[];
```

Comment:
How would something like the `auth-backend` use this? Would we be supporting regex?

Reply:
Yep, basically that